gitlabform/lists/projects.py

```diff
@@ -50,27 +50,19 @@ def _get_single_project(self, target: str) -> Projects:
 
         except NotFoundException:
             debug("Could not find '%s'", target)
-            debug(
-                "Checking if it's a project that needs to be transferred from elsewhere"
-            )
-            project_transfer_source = self.configuration.config["projects_and_groups"][
+            if project_transfer_source := self._get_project_transfer_source_from_config(
                 target
-            ]["project"]["transfer_from"]
-
-            if project_transfer_source:
-                try:
-                    maybe_project = self.gitlab.get_project_case_insensitive(
-                        project_transfer_source
-                    )
-                    debug(
-                        "Found a project '%s' to be transferred to '%s'",
-                        maybe_project["path_with_namespace"],
-                        target,
-                    )
+            ):
+                if self._find_project_transfer_source_in_gitlab(
+                    project_transfer_source
+                ):
                     projects.add_requested([target])
-                except NotFoundException:
-                    # it's a group or a subgroup - ignore it here
-                    pass
+                else:
+                    fatal(
+                        f"""Configuration contains project {target} to be transferred from {project_transfer_source}
+                            but the source project cannot be found in GitLab!""",
+                        exit_code=EXIT_INVALID_INPUT,
+                    )
```

Comment:
Previously/currently if the project's transfer source wasn't found in GitLab, we didn't do a fatal log or set an exit code.

https://github.com/gitlabform/gitlabform/blob/fe8d11e015107ece9291767f29e496405c05228e/gitlabform/lists/projects.py#L60-L73

Do you think that was incorrect? Based on the old code comment in that scenario ("it's a group or a subgroup - ignore it here"), wondering if we should keep it that way. To be honest, that comment is not clear to me.

Reply:
I think it was incorrect. Looking at the chain of logic that leads to this method, `_get_single_project` should only run when the target is a project, so I don't think we should assume it's a group or subgroup if it's not found.

Reply:
Okay, I think that makes sense. I think this may have been overlooked when the code was refactored after this commit.

https://github.com/gitlabform/gitlabform/blob/486132b7b1233076b5f5a52454b523b846f0e6a1/gitlabform/input.py